examples/js/exporters/GLTFExporter.js

```diff
@@ -707,16 +708,27 @@ THREE.GLTFExporter.prototype = {
 
 				var canvas = cachedCanvas = cachedCanvas || document.createElement( 'canvas' );
 
-				canvas.width = image.width;
-				canvas.height = image.height;
+				canvas.width = Math.min( image.width, options.maxTextureSize );
+				canvas.height = Math.min( image.height, options.maxTextureSize );
 
-				if ( options.forcePowerOfTwoTextures && ! isPowerOfTwo( image ) ) {
+				if ( options.forcePowerOfTwoTextures) {
 
-					console.warn( 'GLTFExporter: Resized non-power-of-two image.', image );
+					if(! isPowerOfTwo( image )) {
+						console.warn( 'GLTFExporter: Resized non-power-of-two image.', image );
+					}
 
-					canvas.width = THREE.Math.floorPowerOfTwo( canvas.width );
-					canvas.height = THREE.Math.floorPowerOfTwo( canvas.height );
+					if(! THREE.Math.isPowerOfTwo( canvas.width )) {
+						console.warn( 'GLTFExporter: power-of-two textures requested, but the given maxTextureSize is not power-of-two.')
+					}
```

Comment:
I don't think this requires a warning – we should just use the largest size that is both less than maxTextureSize and still a power of two. We also don't know here that the canvas width came from maxTextureSize. The `THREE.Math.ceilPowerOfTwo( value )` and `THREE.Math.floorPowerOfTwo( value )` functions may be helpful here.

Reply:
You're warning that the user gave a non-POT `maxTextureSize`, but you're checking the value of `canvas.width` and it came from `Math.min( image.width, options.maxTextureSize );`. So, it's possible that the user requested a POT max size, but the actual image was already non-POT and less than that. Still, let's just omit the warning.

And ok, you're using floorPowerOfTwo anyway and ceilPowerOfTwo isn't needed, so that's fine.